### ISSUE TITLE
Doesnt work with modified stub

### ISSUE THREAD

**Issue by AndyWatterman** [lordmulder/rich-header-eraser#1]
Mapping binary file into memory... OK
Scanning for PE header... OK
Scanning for Rich (DanS) header... Failed!

Doesnt work with custom stub, when link any C/C++... projects with option /STUB. However, it should work cuz all pointers are in appropriate places.

This works:

```
//static const DWORD DOS_STUB_LEN = 0x80;
static const DWORD DOS_STUB_LEN = 
	offsetof(IMAGE_DOS_HEADER, e_lfanew) + sizeof((IMAGE_DOS_HEADER*)0)->e_lfanew;
```

**Comment by lordmulder** [comment]
Can you please provide an example file? And exact instruction how it was compiled?

Regards.

**Comment by AndyWatterman** [comment]
Stub file example (just copy-paste to 'stub.com'):

`4d5a9000 03000000 04000000 ffff0000 b8000000 00000000 40000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 40000000`

MSVC 2019 Linker additional option:

`/STUB:stub.com`

Probably, you also need to add to linker cmd  (not sure):

`/filealign:16 /align:16`

The issue is you can make any custom DOS stub with different length (yeah, even with customization VC inserts its Rich header after). But you are using 0x80 as const.

**Comment by lordmulder** [comment]
The MZ file header (struct IMAGE_DOS_HEADER) is 64 bytes in size. After that follows the "DOS stub" code, and then the "Rich" header. So, yeah, I think the problem with _non-standard_ DOS stubs can simply be solved by starting the search for the "Rich" header at address 0x40 instead of at 0x80. Not quite sure what would be the smallest possible "DOS stub" size, but we know for sure that the "Richer" header cannot be located _before_ address 0x40. And certainly not _after_ the PE header start address.

There'd be a slight chance for _false positive_, i.e. to encounter the string "Rich" in "DOS stub" code. But probably can be ignored?

**BTW:** I think `offsetof(IMAGE_DOS_HEADER, e_lfanew) + sizeof((IMAGE_DOS_HEADER*)0)->e_lfanew` is equivalent to `sizeof(IMAGE_DOS_HEADER)`, because "e_lfanew" is the **_last_** field anyway. Hence, offset of "e_lfanew" plus size of "e_lfanew" gives you nothing but the size of the whole struct. And `sizeof(IMAGE_DOS_HEADER)` is simply 64 (0x40).

Regards.

**Comment by lordmulder** [comment]
Should be fixed in f938509d6765d1cd4ef8f64c6a68026840ae4bd7.
